ee/tabby-webserver/src/oauth/github.rs

```diff
@@ -0,0 +1,91 @@
+use anyhow::Result;
+use hyper::{client::HttpConnector, Client, Method, Request};
+use hyper_rustls::HttpsConnector;
+use serde::Deserialize;
+
+use crate::schema::auth::OAuthCredential;
+
+#[derive(Debug, Deserialize)]
+struct GithubOAuthResponse {
+    #[serde(default)]
+    access_token: String,
+    #[serde(default)]
+    scope: String,
+    #[serde(default)]
+    token_type: String,
+
+    #[serde(default)]
+    error: String,
+    #[serde(default)]
+    error_description: String,
+    #[serde(default)]
+    error_uri: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct GithubUserEmail {
+    email: String,
+    primary: bool,
+    verified: bool,
+    visibility: String,
+}
+
+async fn exchange_access_token(
```

Comment:
is it better to wrap an GithubClient around the Client<HttpsClient<...>>, and implement fetch_user_email as its member function?